packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+    * @return totalActiveBalance Total amount of juror tokens active in the Court at the corresponding period checkpoint
+    */
+    function getPeriodBalanceDetails(uint256 _periodId)
+        external
+        view
+        returns (uint64 balanceCheckpoint, uint256 totalActiveBalance)
+    {
+        Period storage period = periods[_periodId];
+        balanceCheckpoint = period.balanceCheckpoint;
+        totalActiveBalance = period.totalActiveBalance;
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror querying the owed fees of
+    * @param _token Address of the token to be queried
+    * @return Token amount corresponding to the juror
+    */
+    function getJurorFees(uint256 _periodId, address _juror, address _token) external view returns (uint256) {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        uint256 totalActiveBalance = period.totalActiveBalance;
+        require(totalActiveBalance != 0, ERROR_PERIOD_BALANCE_DETAILS_NOT_COMPUTED);
+
+        uint256 jurorActiveBalance = _getJurorActiveBalance(_juror, period.balanceCheckpoint);
+        return _getJurorFees(period, _token, jurorActiveBalance, totalActiveBalance);
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror querying the owed fees of
+    * @param _tokens List of token addresses to be queried
+    * @return List of token amounts corresponding to the juror
+    */
+    function getManyJurorFees(uint256 _periodId, address _juror, address[] calldata _tokens) external view returns (uint256[] memory amounts) {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        uint256 totalActiveBalance = period.totalActiveBalance;
+        require(totalActiveBalance != 0, ERROR_PERIOD_BALANCE_DETAILS_NOT_COMPUTED);
+
+        amounts = new uint256[](_tokens.length);
+        uint256 jurorActiveBalance = _getJurorActiveBalance(_juror, period.balanceCheckpoint);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            amounts[i] = _getJurorFees(period, _tokens[i], jurorActiveBalance, totalActiveBalance);
+        }
+    }
+
+    /**
+    * @dev Check if a given juror has already claimed the owed fees for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror being queried
+    * @param _token Address of the token to be queried
+    * @return True if the juror has already claimed the corresponding token fees
+    */
+    function hasJurorClaimed(uint256 _periodId, address _juror, address _token) external view returns (bool) {
```

Comment:
Curious if what you think about flipping this around to `canGuardianClaim()`?

Then we could also check if the specified guardian has any tokens to claim before answering yes/no, which could make it easier to others to know if they should call the claim.

Reply:
Addressed in 55bcbf4056e85ce958a64136d44ad7ea0038c304